arangod/VocBase/vocbase.cpp

```diff
@@ -2022,6 +2097,93 @@ TRI_voc_rid_t TRI_StringToRid(char const* p, size_t len, bool& isOld, bool warn)
   return HybridLogicalClock::decodeTimeStamp(p, len);
 }
 
+
+VocbaseOptions arangodb::getVocbaseOptions(VPackSlice const& options) {
+  // Invalid options will be silently ignored. Default values will be used
+  // instead.
+  //
+  // This Function will be called twice - the second time we do not run in
+  // the risk of consulting the ClusterFeature, because defaults are provided
+  // during the first call.
+
+  std::uint32_t replicationFactor =  1;   // real default will be read
+  std::uint32_t minReplicationFactor = 1; // from ClusterFeature if possible
+  std::string sharding;
+
+  //  sanitize input for vocbase creation
+  //  sharding -- must be "", "flexible" or "single"
+  //  replicationFactor must be "satellite" or a natural number
+  //  minReplicationFactor must be or a natural number
+  {
+    auto shardingSlice = options.get(StaticStrings::Sharding);
+    if(! (shardingSlice.isString()  &&
+          (shardingSlice.compareString("") == 0 || shardingSlice.compareString("flexible") == 0 || shardingSlice.compareString("single") == 0)
+         )) {
+      shardingSlice = VPackSlice::noneSlice();
+    } else {
+      sharding = shardingSlice.copyString();
+    }
+  }
+
+  ClusterFeature* cluster = dynamic_cast<ClusterFeature*>(application_features::ApplicationServer::lookupFeature("Cluster"));
```

Comment:
ClusterFeature should always be there.
You can use `getFeature()`, which will throw if the feature cannot be found. This removes a later if.

Reply:
not true for tests